libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFAuthenticationManager.m

```diff
@@ -846,6 +846,18 @@ - (void)setupWithCredentials:(SFOAuthCredentials*) credentials {
     self.idCoordinator.delegate = self;
 }
 
+- (void)restartAuthentication {
+    @synchronized (self.authBlockList) {
+        if (!self.authenticating) {
+            [self log:SFLogLevelWarning format:@"%@: Authentication manager is not currently authenticating.  No action taken.", NSStringFromSelector(_cmd)];
+            return;
+        }
+        [self log:SFLogLevelInfo format:@"%@: Restarting in-progress authentication process.", NSStringFromSelector(_cmd)];
+        [self.coordinator stopAuthentication];
+        [self loginWithCredentials:[self createOAuthCredentials]];
```

Comment:
Ideally, what I'd like to see is some sort of `SFLoginHostManager` that's completely independent where the value is pulled from. In this case, for now, in `- (void)loginWithCredentials:(SFOAuthCredentials *) credentials`, could we store `credentials` in a property that can be used in `restartAuthentication`? The property in question would be set only when auth is in flight and unset when it completes. It would also be internal, so external consumers can't misuse the property. Would that work?